### ISSUE TITLE
[BUG] Assess on a Map trace throws `no attribute 'get_index'`

### ISSUE THREAD

**Issue by midfield** [genjax-community/genjax#667]
**To Reproduce**
```python
import jax.numpy as jnp
import jax.random as jr; key = jr.PRNGKey(0)
import genjax

fn = genjax.Map(genjax.bernoulli, in_axes=(0,))
trace = fn.simulate(key, (jnp.full((3,), .1),))
fn.assess(key, trace.get_choices(), trace.get_args())

==>

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<@beartype(genjax._src.generative_functions.combinators.vector.map_combinator.MapCombinator.assess) at 0x7fd468be8400>", line 69, in assess
  File "/home/leben_google_com/genjax/src/genjax/_src/generative_functions/combinators/vector/map_combinator.py", line 479, in assess
    check = jnp.count_nonzero(indices - chm.get_index()) == 0
                                        ^^^^^^^^^^^^^
AttributeError: 'VectorChoiceMap' object has no attribute 'get_index'
```

**Execution environment (please complete the following information):**
 - GenJAX version: nightly

**Additional context**
See #570

**Comment by midfield** [comment]
Note that `fn.importance(key, trace.get_choices(), trace.get_args())` also throws an error.

**Comment by sritchie** [comment]
Fixed on nightly, look for 0.4.0 release coming soon.
